### ISSUE TITLE
Replace file() with open() in twisted/conch

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#8526]
|[<img alt="rodrigc's avatar" src="https://avatars.githubusercontent.com/u/1895943?s=50" width="50" height="50">](https://github.com/rodrigc)| @rodrigc reported|
|-|-|
|Trac ID|trac#8526|
|Type|enhancement|
|Created|2016-06-28 06:18:14Z|
|Branch|https://github.com/twisted/twisted/tree/8526-rodrigc-file|

In Python 3, file() is gone (see https://docs.python.org/3/whatsnew/3.0.html#builtins ), and needs to be replaced with open()
which works on Python 2 and 3

<details><summary>Searchable metadata</summary>

```
trac-id__8526 8526
type__enhancement enhancement
reporter__rodrigc rodrigc
priority__normal normal
milestone__Python_3_x Python-3.x
branch__8526_rodrigc_file 8526-rodrigc-file
branch_author__ 
status__closed closed
resolution__fixed fixed
component__core core
keywords__None None
time__1467094694332877 1467094694332877
changetime__1468799508389375 1468799508389375
version__None None
owner__rodrigc rodrigc

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="rodrigc's avatar" src="https://avatars.githubusercontent.com/u/1895943?s=50" width="50" height="50">](https://github.com/rodrigc)<a name="note_1"></a>|@rodrigc commented|
|-|-|

See:
https://github.com/twisted/twisted/pull/250

**Comment by twisted-trac** [comment]
|[<img alt="Lukasa's avatar" src="https://avatars.githubusercontent.com/u/1382556?s=50" width="50" height="50">](https://github.com/Lukasa)<a name="note_3"></a>|@Lukasa commented|
|-|-|

This code looks good, coverage is a concern. The uncovered lines are in conch, you'll need a reviewer who is able either to override the limitation on the merge or who will help write the tests for conch.

**Comment by twisted-trac** [comment]
|[<img alt="adiroiban's avatar" src="https://avatars.githubusercontent.com/u/204609?s=50" width="50" height="50">](https://github.com/adiroiban)<a name="note_4"></a>|@adiroiban commented|
|-|-|

Let's deprecate recvline.Logging and insults.log .... in another ticket.

I am writing tests for the other missing coverage.

**Comment by twisted-trac** [comment]
|[<img alt="adiroiban's avatar" src="https://avatars.githubusercontent.com/u/204609?s=50" width="50" height="50">](https://github.com/adiroiban)<a name="note_5"></a>|@adiroiban set owner to @rodrigc|
|-|-|

So.. cftp script is ugly and I think that the coverage should be done in a separate ticket, together with document how it should behave

I have created [#8542](https://github.com/twisted/twisted/issues/8542) for cftp coverage

Please revert the changes related to cftp.

I have pushed a commit which should fix some coverage for stdio.

For recvline.Logging and insults.log please also revert the changes and we should deprecate them without porting them to py3.

Thanks!

**Comment by twisted-trac** [comment]
|<img alt="GitHub's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50"><a name="note_6"></a>|GitHub set status to `closed`|
|-|-|

In changeset d1dcd8e50822d2a408cf1bbacab16614a6889db3
```
#!CommitTicketReference repository="" revision="d1dcd8e50822d2a408cf1bbacab16614a6889db3"
Merge pull request #250 from twisted/8526-rodrigc-file

Author: rodrigc
Reviewer: alex
Fixes: #8526

Replace file() with open() in twisted/conch
```
